pages/pil/dashboard/index.js

```diff
@@ -22,7 +23,8 @@ module.exports = settings => {
       opts.json.data.establishment = req.model.establishment;
     }
 
-    return req.api(`/establishment/${req.establishmentId}/profiles/${req.profileId}/pil/${req.pilId}/${action}`, opts);
+    return req.api(`/establishment/${req.establishmentId}/profiles/${req.profileId}/pil/${req.pilId}/${action}`, opts)
+      .then(saveTaskIdToSession(req.session));
```

Comment:
this could do with moar `.catch` - I know it wasn't there before, but while you're in there :)

Reply:
This is called from the router below, which has a `catch()` - that should be good enough right?

Reply:
I didn't look that far down, as long as there's a catch somewhere in the promise chain then it shouldn't hang on error